thoth/storages/graph/postgres.py

```diff
@@ -4535,37 +4572,184 @@ def sync_adviser_result(self, document: dict) -> None:
                 is_external=True,
             )
 
-            adviser_run, _ = AdviserRun.get_or_create(
-                session,
-                additional_stack_info=bool(document["result"].get("stack_info")),
-                advised_configuration_changes=bool(document["result"].get("advised_configuration")),
-                adviser_document_id=adviser_document_id,
-                adviser_error=document["result"]["error"],
-                adviser_name=document["metadata"]["analyzer"],
-                adviser_version=document["metadata"]["analyzer_version"],
-                count=parameters["count"],
-                datetime=document["metadata"]["datetime"],
-                debug=cli_arguments.get("verbose", False),
-                duration=document["metadata"].get("duration"),
-                limit=parameters["limit"],
-                limit_latest_versions=parameters.get("limit_latest_versions"),
-                origin=origin,
-                is_s2i=is_s2i,
-                recommendation_type=parameters["recommendation_type"].upper(),
-                requirements_format=parameters["requirements_format"].upper(),
-                external_hardware_information_id=external_hardware_info.id,
-                external_build_software_environment=None,
-                external_run_software_environment_id=external_run_software_environment.id,
-                user_software_stack_id=software_stack.id,
-            )
-
             # Output stacks - advised stacks
             if not document["result"].get("report", {}):
                 _LOGGER.warning(
                     "No report found in %r", adviser_document_id
                 )
+                adviser_run, _ = AdviserRun.get_or_create(
+                    session,
+                    additional_stack_info=bool(document["result"].get("stack_info")),
```

Comment:
To reduce copying this 4 times, you can create a dict and use kwargs to propagate values:

```python
attributes = {
  "additional_stack_info": bool(document["result"].get("stack_info")),
  "advised_configuration_changes": bool(document["result"].get("advised_configuration")),
  # and so on...
}
```

```
AdviserRun.get_or_create(**properties)
```

Or, if you want to distinguish cases where it changes:

```
 AdviserRun.get_or_create(**attributes, need_re_run=False)  # need_re_run must not be in attributes
```

Another solution would be to add items to `attributes` conditionally:

```python
if unresolved_packages:
    attributes["need_re_run"] = True
```
    


IMHO it will reduce code size and could be less error prone.